core/modules/exact_match.cc

```diff
@@ -120,128 +93,75 @@ CommandResponse ExactMatch::Init(const bess::pb::ExactMatchArg &arg) {
 
   for (auto i = 0; i < arg.fields_size(); ++i) {
     CommandResponse err;
-    struct EmField *f = &fields_[i];
-
-    f->pos = size_acc;
 
     if (arg.masks_size() == 0) {
       bess::pb::FieldData emptymask;
-      err = AddFieldOne(arg.fields(i), emptymask, f, i);
+      err = AddFieldOne(arg.fields(i), emptymask, i);
     } else {
-      err = AddFieldOne(arg.fields(i), arg.masks(i), f, i);
+      err = AddFieldOne(arg.fields(i), arg.masks(i), i);
     }
 
     if (err.error().code() != 0) {
       return err;
     }
-
-    size_acc += f->size;
   }
 
   default_gate_ = DROP_GATE;
-  num_fields_ = arg.fields_size();
-  total_key_size_ = align_ceil(size_acc, sizeof(uint64_t));
 
   return CommandSuccess();
 }
 
 void ExactMatch::ProcessBatch(bess::PacketBatch *batch) {
   gate_idx_t default_gate;
-  gate_idx_t out_gates[bess::PacketBatch::kMaxBurst];
-
-  em_hkey_t keys[bess::PacketBatch::kMaxBurst] __ymm_aligned;
+  gate_idx_t out_gates[bess::PacketBatch::kMaxBurst] __ymm_aligned;
```

Comment:
Oops, forgot to note, I don't think this line needs the `__ymm_aligned`, does it? (Not that it really hurts either)